src/main/java/mate/academy/onlinebookstore/service/BookServiceImpl.java

```diff
@@ -37,4 +39,22 @@ public BookDto createBook(CreateBookRequestDto createBookRequestDto) {
         bookRepository.save(book);
         return bookMapper.toDto(book);
     }
+
+    @Override
+    public void deleteById(Long id) {
+        bookRepository.deleteById(id);
+    }
+
+    public BookDto updateBook(Long id, UpdateBookRequestDto updateRequest) {
+        Book book = bookRepository.findById(id)
+                .orElseThrow(() ->
+                        new EntityNotFoundException("Book with id " + id + " not found"));
+        Optional.ofNullable(updateRequest.getTitle()).ifPresent(book::setTitle);
+        Optional.ofNullable(updateRequest.getAuthor()).ifPresent(book::setAuthor);
+        Optional.ofNullable(updateRequest.getIsbn()).ifPresent(book::setIsbn);
+        Optional.ofNullable(updateRequest.getPrice()).ifPresent(book::setPrice);
+        Optional.ofNullable(updateRequest.getDescription()).ifPresent(book::setDescription);
+        Optional.ofNullable(updateRequest.getCoverImage()).ifPresent(book::setCoverImage);
```

Comment:
use the method of mapper 
`void updateBook(Book book, UpdateBookRequestDto updateRequest)`